apps/project-renderer/src/components/app/index.js

```diff
@@ -76,34 +80,32 @@ const App = ( {
 			form.append( key, data[ key ] )
 		);
 
-		return (
-			window
-				.fetch(
-					`https://api.crowdsignal.com/v4/projects/${ projectCode }/form`,
-					{
-						method: 'POST',
-						body: form,
-					}
-				)
-				.then( ( res ) => {
-					if ( ! res.ok ) {
-						throw new Error( res.status );
-					}
-
-					return res.json();
-				} )
-				.then( ( json ) => {
-					if ( ! json || ! json.content ) {
-						throw new Error( 'Empty response' );
-					}
-
-					setContent( json.content );
-					setHasResponded( true );
-					// all the setters should be called here: page, responseHash, content and startTime
-				} )
-				// eslint-disable-next-line no-console
-				.catch( ( err ) => console.error( err ) )
-		);
+		return window
+			.fetch(
+				`https://api.crowdsignal.com/v4/projects/${ projectCode }/form`,
+				{
+					method: 'POST',
+					body: form,
+				}
+			)
+			.then( ( res ) => {
+				if ( ! res.ok ) {
+					throw new Error( res.status );
+				}
+
+				return res.json();
+			} )
+			.then( ( json ) => {
+				if ( ! json || ! json.content ) {
+					throw new Error( 'Empty response' );
+				}
+
+				const responseWithErrors = ( json.errors || [] ).length > 0;
```

Comment:
Error responses should ideally return `400` and that's how we can detect them?
Is there anything that's preventing that? Like, the json responses aren't used anywhere else anyway right?